### ISSUE TITLE
Migrate Discourse from bare-metal to cloud hosting

### ISSUE THREAD

**Issue by jwflory** [FOSSRIT/infrastructure#40]
# Summary

Migrate existing Discourse installation to a cloud provider like Google Cloud / Amazon Web Services

# Background

Currently the application is running on a bare metal server with more resources than it needs. Running Discourse in the cloud helps the site scale better as it grows and also is more economical purchase than our current fixed pool of resources.

# Details

1. Backup existing data (easy to do in Discourse web interface)
2. Set up new cloud instance in cloud platform (I'm leaning towards Google)
3. Install Discourse in cloud environment
4. Import database / config dumps into new install
5. Change DNS records, leave old site available for ~1 week
6. Destroy Discourse site on current server once data and performance are guaranteed

# Outcome

* Easier to scale / get new resources if the site grows
* More economical than renting a VPS with fixed pool of resources

**Comment by jwflory** [comment]
Existing cloud providers offer Discourse site deployments that break from the instructions provided by upstream. I need to do additional research on this before committing to a solution. Since the DigitalOcean droplets are the supported deployment method, I'm considering to migrate to DigitalOcean for this. I'll update again when I explore this in more detail.

**Comment by jwflory** [comment]
I ended up having more time than I anticipated today to work on this specifically. I migrated the application to a DigitalOcean droplet. I aimed to simplify the ability to host this application as low-maintenance as possible (thinking ahead to when I'm no longer administering this site).

The migration work is done, but now need to figure out billing with RIT MAGIC Center. Estimated monthly cost looks like $6 a month. Includes automatic weekly backups. Marking ticket as blocked until billing is resolved. Then this ticket can be closed.

**Comment by jwflory** [comment]
To give an update, a new DigitalOcean account is opened by the MAGIC Center (specifically by Brenda). I opened a DigitalOcean support ticket for help migrating the droplet from my account into MAGIC's. There is some additional work I need to do to successfully migrate to the new account. I hope to have this done by next Monday at the latest.

**Comment by jwflory** [comment]
This is now complete. :clapper: Closing.

The DigitalOcean Droplet with the fossrit.community Discourse site is managed under a DigitalOcean Team. The team membership includes me and Brenda as owners. @itprofjacobs was sent an invitation as well. I migrated a snapshot of the Discourse site from my account to the new one, created a new Droplet, and tested the site with @whenbellstoll. I also enabled automated weekly backups on the DigitalOcean side, coming in at an extra $2/month.
